compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/propagate-scope-deps-hir-fork/reduce-reactive-deps/merge-uncond-optional-chain-and-cond.expect.md

```diff
@@ -0,0 +1,92 @@
+
```

Comment:
This is an extreme edge case and not code we'd expect any reasonable developer to write. In most cases (`a?.b != null ? a.b : DEFAULT`), we do want to take a dependency on `a?.b`.

I found this trying to come up with edge cases that break the current dependency + CFG merging logic. I think it makes sense to error on the side of correctness. After all, we still take `a` as a dependency if users write `a != null ? a.b : DEFAULT`, and the same fix (understanding the hoistable `!= null` test expression) works for both. Can be convinced otherwise though!